lib/plugins/aws/invoke.js

```diff
@@ -22,26 +22,34 @@ class AwsInvoke {
     };
   }
 
+  validateFile(key) {
+    const absolutePath = path.isAbsolute(this.options[key])
+      ? this.options[key]
+      : path.join(this.serverless.config.servicePath, this.options[key]);
+    if (!this.serverless.utils.fileExistsSync(absolutePath)) {
+      throw new this.serverless.classes.Error('The file you provided does not exist.');
+    }
+    return this.serverless.utils.readFileSync(absolutePath);
+  }
+
   async extendedValidate() {
     this.validate();
     // validate function exists in service
     this.options.functionObj = this.serverless.service.getFunction(this.options.function);
-    this.options.data = this.options.data || '';
-
-    if (!this.options.data) {
-      if (this.options.path) {
-        const absolutePath = path.isAbsolute(this.options.path)
-          ? this.options.path
-          : path.join(this.serverless.config.servicePath, this.options.path);
-        if (!this.serverless.utils.fileExistsSync(absolutePath)) {
-          throw new this.serverless.classes.Error('The file you provided does not exist.');
-        }
-        this.options.data = this.serverless.utils.readFileSync(absolutePath);
-      } else {
-        try {
-          this.options.data = await stdin();
-        } catch {
-          // continue if no stdin was provided
+    for (const [rawDataKey, pathKey] of [
+      ['data', 'path'],
+      ['context', 'contextPath'],
+    ]) {
+      this.options[rawDataKey] = this.options[rawDataKey] || '';
+      if (!this.options[rawDataKey]) {
+        if (this.options[pathKey]) {
+          this.options[rawDataKey] = this.validateFile(pathKey);
+        } else {
+          try {
+            this.options[rawDataKey] = await stdin();
```

Comment:
This should be applied only to `data` resolution. 

I think this means that using `for .. of` as proposed is probably not best choice